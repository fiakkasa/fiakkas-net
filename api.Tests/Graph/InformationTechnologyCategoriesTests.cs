using HotChocolate.Execution;

namespace api.Tests.Graph;

public class InformationTechnologyCategoriesTests(GraphFixture fixture) : IClassFixture<GraphFixture>
{
    [Fact]
    public async Task InformationTechnologyCategories_Should_Return_Data()
    {
        var executor = await fixture.GetRequestExecutor();

        var result = await executor.ExecuteAsync(
            """
            {
              informationTechnologyCategories {
                totalCount
                nodes {
                  createdAt
                  href
                  id
                  internalId
                  title
                  updatedAt
                  version
                  portfolioCategories {
                    totalCount
                  }
                  portfolioCustomers {
                    totalCount
                  }
                  portfolioItems {
                    totalCount
                  }
                }
              }
            }
            """);

        var fn = result.ExpectOperationResult;

        Assert.Null(fn().Errors);
        result.ToJson().MatchSnapshot();
    }

    [Fact]
    public async Task InformationTechnologyCategoryById_Should_Return_Data()
    {
        var executor = await fixture.GetRequestExecutor();

        var result = await executor.ExecuteAsync(
            """
            {
              informationTechnologyCategoryById(id: "SW5mb3JtYXRpb25UZWNobm9sb2d5Q2F0ZWdvcnkKZzZjYzQzZTlhMzEyYjQ5MjNiODkwZTk2NmI4MTY4ZWVl") {
                createdAt
                href
                id
                internalId
                title
                updatedAt
                version
              }
            }
            """);

        var fn = result.ExpectOperationResult;

        Assert.Null(fn().Errors);
        result.ToJson().MatchSnapshot();
    }
}
