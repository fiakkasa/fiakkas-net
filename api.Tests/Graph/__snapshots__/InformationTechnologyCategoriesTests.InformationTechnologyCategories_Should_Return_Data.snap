{
  "data": {
    "informationTechnologyCategories": {
      "totalCount": 1,
      "nodes": [
        {
          "createdAt": "2024-01-01T00:00:00.000Z",
          "href": "/test",
          "id": "SW5mb3JtYXRpb25UZWNobm9sb2d5Q2F0ZWdvcnkKZzZjYzQzZTlhMzEyYjQ5MjNiODkwZTk2NmI4MTY4ZWVl",
          "internalId": "6cc43e9a-312b-4923-b890-e966b8168eee",
          "title": "Title",
          "updatedAt": null,
          "version": 1,
          "portfolioCategories": {
            "totalCount": 0
          },
          "portfolioCustomers": {
            "totalCount": 0
          },
          "portfolioItems": {
            "totalCount": 0
          }
        }
      ]
    }
  }
}
